src/lib/utils.ts

```diff
@@ -1,6 +1,16 @@
-import { clsx, type ClassValue } from "clsx"
-import { twMerge } from "tailwind-merge"
+import { clsx, type ClassValue } from "clsx";
+import { twMerge } from "tailwind-merge";
+import { client } from "@/lib/thirdwebClient";
+import { getCurrencyMetadata } from "thirdweb/extensions/erc20";
+import { getContract } from "thirdweb";
+import { defaultChain } from "@/lib/constants";
 
 export function cn(...inputs: ClassValue[]) {
-  return twMerge(clsx(inputs))
+  return twMerge(clsx(inputs));
+}
+
+export async function fetchCurrencyMetadata(currency: string) {
```

Comment:
Also, you don't need this function 

Just use getCurrencyMetadata directly instead of this wrapper